### PR TITLE
Moving api -> apis and setting multigroup: true

### PR DESCRIPTION
**Pull request by christopherhein**
Once we have `NestedCluster` we're going to need to have a `multigroup` controllers turned on to save a bunch of pain later this moved NCP into `apis/controlplanes`. This also brings the Dockerfile from a standard `kubebuilder` project.

Signed-off-by: Chris Hein <me@chrishein.com>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -56,8 +60,9 @@ var (
 func init() {
 	klog.InitFlags(nil)
 
-	_ = clientgoscheme.AddToScheme(scheme)
-	_ = clusterv1.AddToScheme(scheme)
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(clusterv1.AddToScheme(scheme))
+	utilruntime.Must(controlplanev1alpha4.AddToScheme(scheme))
```
Comment (christopherhein) [inline main.go:65]: `utilruntime.Must()` will fail if the types aren't loaded.

**Review by Fei-Guo — APPROVED** [review]

**Comment by Fei-Guo** [comment]
@christopherhein Please rebase to resolve the conflict.

**Comment by christopherhein** [comment]
Thanks @Fei-Guo all updated. Just needs. `lgtm`

**Pull request merged**